src/constbuffer_array.c

```diff
@@ -281,19 +282,29 @@ IMPLEMENT_MOCKABLE_FUNCTION(, CONSTBUFFER_ARRAY_HANDLE, constbuffer_array_create
                         {
                             if (i == 0)
                             {
-                                result->buffers[i] = start_buffer;
+                                result->buffers[i - empty_buffer_count] = start_buffer;
                             }
                             else if (i == buffer_count - 1)
                             {
-                                result->buffers[i] = end_buffer;
+                                result->buffers[i - empty_buffer_count] = end_buffer;
                             }
                             else
                             {
-                                /* Codes_SRS_CONSTBUFFER_ARRAY_07_008: [ constbuffer_array_create_from_start_and_end shall copy all of the CONSTBUFFER_HANDLES except first and last buffer from each const buffer array in buffer_arrays to the newly constructed array by calling CONSTBUFFER_IncRef. ]*/
-                                CONSTBUFFER_IncRef(original->buffers[start_buffer_index + i]);
-                                result->buffers[i] = original->buffers[start_buffer_index + i];
+                                /* Codes_SRS_CONSTBUFFER_ARRAY_07_015: [ constbuffer_array_create_from_start_and_end shall skip any empty buffers in the middle of the original buffer. ]*/
```

Comment:
> /* Codes_SRS_CONSTBUFFER_ARRAY_07_015: [ constbuffe [](http://example.com/codeflow?start=31&length=52)

I don't think we should do this.
This is effectively saying that empty buffers are bad (otherwise why would we skip them?).
If they are bad I would be thinking we should reject them when we create a const buffer array.
Buuuut we do not reject them and maybe there are cases where we actually put them in the array.

I guess the question is why do we care to eliminate the zero buffers in this process of splitting?

Reply:
deleted this